types/mongooseoptions.d.ts

```diff
@@ -0,0 +1,178 @@
+     *
+     * @default true
+     */
+    bufferCommands?: boolean;
+
+    /**
+     * If bufferCommands is on, this option sets the maximum amount of time Mongoose
+     * buffering will wait before throwing an error.
+     * If not specified, Mongoose will use 10000 (10 seconds).
+     *
+     * @default 10000
+     */
+    bufferTimeoutMS?: number;
+
+    /**
+     * Set to `true` to `clone()` all schemas before compiling into a model.
+     *
+     * @default false
+     */
+    cloneSchemas?: boolean;
+
+    /**
+     * If `true`, prints the operations mongoose sends to MongoDB to the console.
+     * If a writable stream is passed, it will log to that stream, without colorization.
+     * If a callback function is passed, it will receive the collection name, the method
+     * name, then all arguments passed to the method. For example, if you wanted to
+     * replicate the default logging, you could output from the callback
+     * `Mongoose: ${collectionName}.${methodName}(${methodArgs.join(', ')})`.
+     *
+     * @default false
+     */
+    debug?:
+    | boolean
+    | { color?: boolean; shell?: boolean; }
+    | stream.Writable
+    | ((collectionName: string, methodName: string, ...methodArgs: any[]) => void);
+
+    /** If set, attaches [maxTimeMS](https://docs.mongodb.com/manual/reference/operator/meta/maxTimeMS/) to every query */
+    maxTimeMS?: number;
+
+    /**
+     * Mongoose adds a getter to MongoDB ObjectId's called `_id` that
+     * returns `this` for convenience with populate. Set this to false to remove the getter.
+     *
+     * @default true
+     */
+    objectIdGetter?: boolean;
+
+    /**
+     * Set to `true` to default to overwriting models with the same name when calling
+     * `mongoose.model()`, as opposed to throwing an `OverwriteModelError`.
+     *
+     * @default false
+     */
+    overwriteModels?: boolean;
+
+    /**
+     * If `false`, changes the default `returnOriginal` option to `findOneAndUpdate()`,
+     * `findByIdAndUpdate`, and `findOneAndReplace()` to false. This is equivalent to
+     * setting the `new` option to `true` for `findOneAndX()` calls by default. Read our
+     * `findOneAndUpdate()` [tutorial](https://mongoosejs.com/docs/tutorials/findoneandupdate.html)
+     * for more information.
+     *
+     * @default true
+     */
+    returnOriginal?: boolean;
+
+    /**
+     * false by default. Set to true to enable [update validators](
+     * https://mongoosejs.com/docs/validation.html#update-validators
+     * ) for all validators by default.
+     */
+    runValidators?: boolean;
+
+    /**
+     * Sanitizes query filters against [query selector injection attacks](
+     * https://thecodebarbarian.com/2014/09/04/defending-against-query-selector-injection-attacks.html
+     * ) by wrapping any nested objects that have a property whose name starts with $ in a $eq.
+     */
+    sanitizeFilter?: boolean;
+
+    sanitizeProjection?: boolean;
```

Comment:
There was no information for sanitizeProjection regarding default value or what it does in the documentation.

Reply:
false by default. There's some more info on sanitizeProjection here: https://thecodebarbarian.com/whats-new-in-mongoose-5-13-sanitizeprojection.html . Should add that to the docs.